stable_baselines3/common/policies.py

```diff
@@ -583,39 +577,21 @@ def forward(self, obs: th.Tensor, deterministic: bool = False) -> Tuple[th.Tenso
         :param deterministic: Whether to sample or use deterministic actions
         :return: action, value and log probability of the action
         """
-        latent_pi, latent_vf, latent_sde = self._get_latent(obs)
+        # Preprocess the observation if needed
+        features = self.extract_features(obs)
+        latent_pi, latent_vf = self.mlp_extractor(features)
         # Evaluate the values for the given observations
         values = self.value_net(latent_vf)
-        distribution = self._get_action_dist_from_latent(latent_pi, latent_sde=latent_sde)
+        distribution = self._get_action_dist_from_latent(latent_pi)
         actions = distribution.get_actions(deterministic=deterministic)
         log_prob = distribution.log_prob(actions)
         return actions, values, log_prob
 
-    def _get_latent(self, obs: th.Tensor) -> Tuple[th.Tensor, th.Tensor, th.Tensor]:
```

Comment:
I see the point of removing this part (only few lines of code), but I wonder if some people rely on this for debugging/modification purposes. Might be better idea to keep it for now for backwards compat (yes, not an "official API" thing, but just a thought :) )

Edit: This function was also used in at least four places, so it might be worth keeping?

Reply:
yes, I do not consider it as part of the API so I'm ok with breaking it.

The main argument against was that it was always computing latent variables for policy and value but most of the time you just need one of the two, hence the removal (it saves computation and makes things clearer by having a better separation).

Reply:
Gotcha!